Completed flicker reduction and refactoring.

g8kig opened the pull request:
I think it is working quite well now.

chillmf commented:
Paul,

Thanks for pulling all of this together.

I will try it out and give you a report tomorrow morning my time.

Have a good night!

Regards,

Charley

I

On Fri, Jan 24, 2025 at 1:56 PM Paul Winwood ***@***.***>
wrote:

> I think it is working quite well now.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>
> https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8
> Commit Summary
>
>    - 3995195
>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/commits/39951950fb5a5ab9592fd1ecc5e5e3afd7168113>
>    Added linux build script
>    - 4299581
>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/commits/429958178f1b6e64d5c56ccc089f9be32ed1ab21>
>    Use the standard path for platformio
>    - 0c39567
>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/commits/0c39567440fb0d9fb31d7a507154e29fbfa72451>
>    Merge branch 'SevenBandTransceiver' of
>    https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source
>    into SevenBandTransceiver
>    - de2d868
>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/commits/de2d86887f6def36c90f34e377cb8a05e7fd2e0f>
>    General refactoring
>    - 8c1cadf
>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/commits/8c1cadf22bcfbd7c6afd1287c6ca6fc81b439c08>
>    More refactoring
>    - 6caba68
>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/commits/6caba688388cd38e46d939613a7a1bc148e0150c>
>    Remove unnecessary files
>    - 770c20c
>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/commits/770c20cb52ba2cc7af258f2ee322cbf18f78fdb4>
>    Refactored the waterfall display
>    - 3d9f59a
>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/commits/3d9f59aa284ea38aa65c6581eec65f72d47c1761>
>    Optimise waterfall drawing
>    - 14da38a
>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/commits/14da38ac91655d52671af4f7adae1d8be6859e63>
>    Scale the FFT datum to the number of entries in the colour lookup table
>    - a93c162
>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/commits/a93c162b96169c4a01b3732ab478aecd2676e767>
>    Remove FillRect as it caused too much flicker
>    - 7c94400
>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/commits/7c944002f1fc089ef211af11503d011968ebe818>
>    Go back to FillRect except draw from the bottom up (slight waterfall
>    flicker)
>    - e541cf4
>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/commits/e541cf4ab3ff83f8c0c06e184742f55e192b74a3>
>    Some more refactoring and FFT display flicker reduction
>
> File Changes
>
> (21 files
> <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files>
> )
>
>    - *M* .cproject
>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-1ebe600e005d9c03f4e213f2ee0ca4c31b4e9df54c4ca3c7a8a352ef6b153ec2>
>    (3)
>    - *R* DSP_CMSIS/arm_bitreversal2.s
>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-3eeb6ef3d209f116f96bf60beedb10bb4c31766065a2dd273266168828f2fa06>
>    (0)
>    - *M* Inc/Display.h
>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-8a42d4b5374dbecc21f27c16c4bce388c634078acea8d5e4689f027a419789e7>
>    (6)
>    - *M* Inc/WF_Table.h
>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-7000ca6ebe5298aef6738bffe3295e9b2a11139e316ef0dc683d52117f1d0eb8>
>    (35)
>    - *M* Inc/button.h
>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-6d58d0b4fc5febfbe7b669e61c3d3e2c17d97d27e528c5dd6b8604977da3eef5>
>    (4)
>    - *M* Inc/decode_ft8.h
>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-4e003662aa650198d570ab7a687b637df291eda34369c4b9665eb2349af418b7>
>    (1)
>    - *M* Inc/gen_ft8.h
>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-3865b92948db498a2bce2f6ac68ef0bb4b4811f3e42001dcea7c86abc3ef8ad5>
>    (6)
>    - *M* Inc/main.h
>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-b495c49aca81c43628e5058dd6ae7da8113bec0e9e75c8bbc91d6a74beabe8db>
>    (7)
>    - *M* Inc/stm32f7xx_it.h
>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-cf624619e1398072c24c0d705b95eb57833d26a7ab8bc59b65d3fd578b963642>
>    (2)
>    - *D* Katy_converter.log
>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-49677125c5c4d33f7954487c2663c980a5ca0f73e73a2eef9995210ef3f6144c>
>    (10)
>    - *D* Src/ChangeLog
>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-62cf7d3380166ff4739fa7c67f787d25b1f36aa7ec7134e5917bd0c3d4cc422e>
>    (0)
>    - *M* Src/Display.c
>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-f7fa23ab2e81054c9bb8c348ebe6fb7fecb475d4ed67ea3690b0318073d9aff0>
>    (150)
>    - *M* Src/button.c
>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-2d01b95f99ee9dc4e07087c74483a9683ba281259ad49e16c5134ab58829773e>
>    (134)
>    - *M* Src/decode_ft8.c
>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-20c23cb4ec62a88d36cd6e6cb9c192e640a76c53da95aface09cd768d6ba726e>
>    (4)
>    - *M* Src/main.c
>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-7e6d8be214f6d3cb74bbe82096d6aefa319804b1cce6ddd6745f1561adb42263>
>    (2)
>    - *M* Src/options.c
>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-5f201aef21dd04a386b67e3142a395ee5a5fbf86dcf59cfb01f697b7433d5d3d>
>    (2)
>    - *D* Utilities/Release_Notes.html
>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-bc4c740f58feab335e7a8f19b189546eb3ae3e06ffe2a66cb4eba16574e7d03d>
>    (163)
>    - *D* Utilities/lcd_log.c
>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-bb9e200cee63059058c967a88950ca44d758fa96470340e29f29bb7a36a69675>
>    (474)
>    - *D* Utilities/lcd_log.h
>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-c5e2665aff82683f22bbc19310cfac6aee66993cd19e0f1b96806e51cb366718>
>    (161)
>    - *M* build.cmd
>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-5d1a8c1f0cf259b0ef0ad38234ee6760764b2eb416f063f62bb7e54bf84d03c7>
>    (285)
>    - *A* build.sh
>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-4d2a8eefdf2a9783512a35da4dc7676a66404b6f3826a8af9aad038722da6823>
>    (152)
>
> Patch Links:
>
>    -
>    https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8.patch
>    -
>    https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8.diff
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAMELXXIQGQQFREDHYDQELD2MKLFDAVCNFSM6AAAAABV2MIJQKVHI2DSMVQWIX3LMV43ASLTON2WKOZSHAYTAMRXG44DINA>
> .
> You are receiving this because your review was requested.Message ID:
> ***@***.***>
>

chillmf commented:
Paul,

I tried the code related to your pull request and I found that there is a
significant flicker. This flicker is a black band going from the bottom of
the display to the top. Also, it is intermittent.

I have made a video to show what I am seeing.

Hopefully you will be able to access the video using this link:

https://drive.google.com/file/d/1kjap_p409M2ktDWwBZnrkB65PE8v5zM1/view?usp=drive_link

If you can not access the video please let me know and I will try another
way.

I tested the buttons as well and everything works ok.

Regards,

Charley



On Fri, Jan 24, 2025 at 3:33 PM Charles Hill ***@***.***> wrote:

> Paul,
>
> Thanks for pulling all of this together.
>
> I will try it out and give you a report tomorrow morning my time.
>
> Have a good night!
>
> Regards,
>
> Charley
>
> I
>
> On Fri, Jan 24, 2025 at 1:56 PM Paul Winwood ***@***.***>
> wrote:
>
>> I think it is working quite well now.
>> ------------------------------
>> You can view, comment on, or merge this pull request online at:
>>
>>
>> https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8
>> Commit Summary
>>
>>    - 3995195
>>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/commits/39951950fb5a5ab9592fd1ecc5e5e3afd7168113>
>>    Added linux build script
>>    - 4299581
>>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/commits/429958178f1b6e64d5c56ccc089f9be32ed1ab21>
>>    Use the standard path for platformio
>>    - 0c39567
>>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/commits/0c39567440fb0d9fb31d7a507154e29fbfa72451>
>>    Merge branch 'SevenBandTransceiver' of
>>    https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source
>>    into SevenBandTransceiver
>>    - de2d868
>>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/commits/de2d86887f6def36c90f34e377cb8a05e7fd2e0f>
>>    General refactoring
>>    - 8c1cadf
>>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/commits/8c1cadf22bcfbd7c6afd1287c6ca6fc81b439c08>
>>    More refactoring
>>    - 6caba68
>>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/commits/6caba688388cd38e46d939613a7a1bc148e0150c>
>>    Remove unnecessary files
>>    - 770c20c
>>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/commits/770c20cb52ba2cc7af258f2ee322cbf18f78fdb4>
>>    Refactored the waterfall display
>>    - 3d9f59a
>>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/commits/3d9f59aa284ea38aa65c6581eec65f72d47c1761>
>>    Optimise waterfall drawing
>>    - 14da38a
>>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/commits/14da38ac91655d52671af4f7adae1d8be6859e63>
>>    Scale the FFT datum to the number of entries in the colour lookup table
>>    - a93c162
>>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/commits/a93c162b96169c4a01b3732ab478aecd2676e767>
>>    Remove FillRect as it caused too much flicker
>>    - 7c94400
>>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/commits/7c944002f1fc089ef211af11503d011968ebe818>
>>    Go back to FillRect except draw from the bottom up (slight waterfall
>>    flicker)
>>    - e541cf4
>>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/commits/e541cf4ab3ff83f8c0c06e184742f55e192b74a3>
>>    Some more refactoring and FFT display flicker reduction
>>
>> File Changes
>>
>> (21 files
>> <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files>
>> )
>>
>>    - *M* .cproject
>>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-1ebe600e005d9c03f4e213f2ee0ca4c31b4e9df54c4ca3c7a8a352ef6b153ec2>
>>    (3)
>>    - *R* DSP_CMSIS/arm_bitreversal2.s
>>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-3eeb6ef3d209f116f96bf60beedb10bb4c31766065a2dd273266168828f2fa06>
>>    (0)
>>    - *M* Inc/Display.h
>>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-8a42d4b5374dbecc21f27c16c4bce388c634078acea8d5e4689f027a419789e7>
>>    (6)
>>    - *M* Inc/WF_Table.h
>>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-7000ca6ebe5298aef6738bffe3295e9b2a11139e316ef0dc683d52117f1d0eb8>
>>    (35)
>>    - *M* Inc/button.h
>>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-6d58d0b4fc5febfbe7b669e61c3d3e2c17d97d27e528c5dd6b8604977da3eef5>
>>    (4)
>>    - *M* Inc/decode_ft8.h
>>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-4e003662aa650198d570ab7a687b637df291eda34369c4b9665eb2349af418b7>
>>    (1)
>>    - *M* Inc/gen_ft8.h
>>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-3865b92948db498a2bce2f6ac68ef0bb4b4811f3e42001dcea7c86abc3ef8ad5>
>>    (6)
>>    - *M* Inc/main.h
>>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-b495c49aca81c43628e5058dd6ae7da8113bec0e9e75c8bbc91d6a74beabe8db>
>>    (7)
>>    - *M* Inc/stm32f7xx_it.h
>>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-cf624619e1398072c24c0d705b95eb57833d26a7ab8bc59b65d3fd578b963642>
>>    (2)
>>    - *D* Katy_converter.log
>>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-49677125c5c4d33f7954487c2663c980a5ca0f73e73a2eef9995210ef3f6144c>
>>    (10)
>>    - *D* Src/ChangeLog
>>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-62cf7d3380166ff4739fa7c67f787d25b1f36aa7ec7134e5917bd0c3d4cc422e>
>>    (0)
>>    - *M* Src/Display.c
>>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-f7fa23ab2e81054c9bb8c348ebe6fb7fecb475d4ed67ea3690b0318073d9aff0>
>>    (150)
>>    - *M* Src/button.c
>>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-2d01b95f99ee9dc4e07087c74483a9683ba281259ad49e16c5134ab58829773e>
>>    (134)
>>    - *M* Src/decode_ft8.c
>>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-20c23cb4ec62a88d36cd6e6cb9c192e640a76c53da95aface09cd768d6ba726e>
>>    (4)
>>    - *M* Src/main.c
>>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-7e6d8be214f6d3cb74bbe82096d6aefa319804b1cce6ddd6745f1561adb42263>
>>    (2)
>>    - *M* Src/options.c
>>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-5f201aef21dd04a386b67e3142a395ee5a5fbf86dcf59cfb01f697b7433d5d3d>
>>    (2)
>>    - *D* Utilities/Release_Notes.html
>>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-bc4c740f58feab335e7a8f19b189546eb3ae3e06ffe2a66cb4eba16574e7d03d>
>>    (163)
>>    - *D* Utilities/lcd_log.c
>>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-bb9e200cee63059058c967a88950ca44d758fa96470340e29f29bb7a36a69675>
>>    (474)
>>    - *D* Utilities/lcd_log.h
>>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-c5e2665aff82683f22bbc19310cfac6aee66993cd19e0f1b96806e51cb366718>
>>    (161)
>>    - *M* build.cmd
>>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-5d1a8c1f0cf259b0ef0ad38234ee6760764b2eb416f063f62bb7e54bf84d03c7>
>>    (285)
>>    - *A* build.sh
>>    <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8/files#diff-4d2a8eefdf2a9783512a35da4dc7676a66404b6f3826a8af9aad038722da6823>
>>    (152)
>>
>> Patch Links:
>>
>>    -
>>    https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8.patch
>>    -
>>    https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8.diff
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/g8kig/DX-FT8-MULTIBAND-TABLET-TRANSCEIVER-Source/pull/8>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AAMELXXIQGQQFREDHYDQELD2MKLFDAVCNFSM6AAAAABV2MIJQKVHI2DSMVQWIX3LMV43ASLTON2WKOZSHAYTAMRXG44DINA>
>> .
>> You are receiving this because your review was requested.Message ID:
>> ***@***.***>
>>
>

g8kig commented:
Sorry Charley, I have made even more changes. Principally, I have added parsing extra data from Station.Txt for "CQ SOTA" but it is not integrated yet. Carriage returns and line feeds are accepted now. I haved also removed the loop between touch down and touch up by using a little state machine. A few more refactorings and clean ups. I am going to stop now!

g8kig commented:
Ok reverted to the original drawing code.

chillmf reviewed (commented):
All is well.
Thanks for the hard work.
